src/Microsoft.AspNetCore.Blazor.Server/Builder/ServerSideBlazorApplicationBuilderExtensions.cs

```diff
@@ -20,94 +19,57 @@ public static class ServerSideBlazorApplicationBuilderExtensions
 
-            if (startupType == null)
-            {
-                throw new ArgumentNullException(nameof(startupType));
-            }
-
-            var startup = builder.ApplicationServices.GetRequiredService(startupType);
-            var wrapper = new ConventionBasedStartup(startup);
-            Action<IBlazorApplicationBuilder> configure = (b) =>
-            {
-                wrapper.Configure(b, b.Services);
-            };
-
-            UseServerSideBlazorCore(builder, configure);
-
-            builder.UseBlazor(new BlazorOptions()
-            {
-                ClientAssemblyPath = startupType.Assembly.Location,
-            });
-
-            return builder;
+            // TODO: Also allow configuring the endpoint path.
+            return UseSignalR(builder, BlazorHub.DefaultPath)
+                .UseBlazor<TStartup>();
         }
 
         /// <summary>
-        /// Registers middleware for Server-Side Blazor.
+        /// Registers Server-Side Blazor in the pipeline.
         /// </summary>
         /// <param name="builder">The <see cref="IApplicationBuilder"/>.</param>
         /// <param name="options">A <see cref="BlazorOptions"/> instance used to configure the Blazor file provider.</param>
-        /// <param name="startupAction">A delegate used to configure the renderer.</param>
         /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
         public static IApplicationBuilder UseServerSideBlazor(
             this IApplicationBuilder builder,
-            BlazorOptions options,
-            Action<IBlazorApplicationBuilder> startupAction)
+            BlazorOptions options)
         {
             if (builder == null)
             {
                 throw new ArgumentNullException(nameof(builder));
             }
 
-            if (startupAction == null)
+            if (options == null)
             {
-                throw new ArgumentNullException(nameof(startupAction));
+                throw new ArgumentNullException(nameof(options));
             }
 
-            UseServerSideBlazorCore(builder, startupAction);
-
-            builder.UseBlazor(options);
+            // WARNING: Don't add extra setup logic here. It's important for
+            // UseServerSideBlazor just to be shorthand for UseSignalR+UseBlazor,
+            // so that people who want to call those two manually instead can
+            // also do so. That's needed for people using Azure SignalR.
 
-            return builder;
+            // TODO: Also allow configuring the endpoint path.
+            return UseSignalR(builder, BlazorHub.DefaultPath)
+                .UseBlazor(options);
         }
 
-        private static IApplicationBuilder UseServerSideBlazorCore(
-            IApplicationBuilder builder,
-            Action<IBlazorApplicationBuilder> configure)
+        private static IApplicationBuilder UseSignalR(IApplicationBuilder builder, PathString path)
```

Comment:
nit: different name? this makes the call sites look like a call to the vanilla UseSignalR, but it's totally not.

Reply:
Good point. Renamed to `UseSignalRWithBlazorHub`.